Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM golang:1.10.1-alpine3.7 as go-builder
+
+ARG PROJECT_NAME=zookeeper-operator
+ARG REPO_PATH=github.com/pravega/$PROJECT_NAME
+ARG BUILD_PATH=${REPO_PATH}/cmd/${PROJECT_NAME}
+
+# Build version and commit should be passed in when performing docker build
+ARG VERSION=0.0.0-localdev
+ARG GIT_SHA=0000000
+
+COPY pkg /go/src/${REPO_PATH}/pkg
```

Comment:
setting up `GO_PATH` and `GO_HOME` and then running `go install` may be the correct way of doing it. Is there any specific reason we are copying the bits by hand?

Reply:
These are the only bits that we need to build the operator. We could copy the entire root directory to reduce the number of lines from 3 to 1, but we would be copying useless data.

Reply:
The `go` cmd `install` (as described here https://nanxiao.gitbooks.io/golang-101-hacks/content/posts/go-build-vs-go-install.html) does exactly the same thing you are trying to accomplish. My question was about the need to do it ourselves.

Reply:
@arvindkandhare I still don't understand how we can benefit from `go install`. AFAIK, `go install` builds the project and moves the binary to a specific location, but we don't need to move the binary anywhere else since we will copy it to the final destination in the next stage. Can you please elaborate a bit more your suggestion? Or we can also discuss it offline.